Rename classes

bushw011 opened the pull request:
It makes a simple call to the database that updates the classname. The UI for this is an edit button next to the class title that only appears for teachers. This button allows them to do an inline (no dialog popup) edit of the name. Closes #39

|![localhost_9000_class_p7w93wrki44pzqxnkiqw iphone 6](https://user-images.githubusercontent.com/25711751/33520931-f0e5639e-d789-11e7-9b41-b5b17b68eafb.png)|
![localhost_9000_class_p7w93wrki44pzqxnkiqw iphone 6 1](https://user-images.githubusercontent.com/25711751/33520932-f3a29e3a-d789-11e7-8db5-f9545314893c.png)|
![localhost_9000_class_p7w93wrki44pzqxnkiqw iphone 6 2](https://user-images.githubusercontent.com/25711751/33520933-f6faddfe-d789-11e7-8ed6-f5eb2522855a.png)|
|---|---|---|

bushw011 committed:
Implement renaming classes

The UI is getting there but the way it's done behind the scenes is a bit gross and could use some refactoring. Also need to utilize a promise on the function call for a success/error message

bushw011 committed:
Get rid of component for renaming that won't be used

bushw011 committed:
Add styling for save class button

This effectively finishes the UI for renaming classes

bushw011 committed:
Add css so the edit button lines up with class name

bushw011 committed:
Add success/error messages

Also the save button is now disabled if the user tries to change the class name to nothing

bushw011 committed:
Fix css bug that made mat-tool-tip appear far off for edit button

floogulinc commented on src/app/class-component/class.component.ts:
```diff
@@ -96,6 +98,19 @@ export class ClassComponent implements OnInit, OnDestroy {
         console.log("class destroyed");
     }
 
+    public updateClass(): void {
```
should probably be called renameClass

floogulinc commented on src/app/class/class.service.ts:
```diff
@@ -114,6 +114,12 @@ export class ClassService {
             });
     }
 
+    public editClass(classId: string, newClassName: string){
```
should probably be called renameClass

bushw011 committed:
Change function names to be more straightforward

bushw011 committed:
Fix mat-tool-tip to say the right thing

EthanUphoff reviewed (dismissed):
Looks good to me, runs well, no bugs found.

floogulinc reviewed (changes requested):
The menu icon shouldn't move when editing a card on mobile as it does in the screenshot.

floogulinc committed:
Correct snackbar text

bushw011 committed:
Make change name button for classes a 'save' icon

bushw011 committed:
Merge remote-tracking branch 'origin/rename-class' into rename-class

plumstyle committed:
fix position of editing class name

floogulinc commented on src/app/class-component/class.component.css:
```diff
@@ -13,3 +13,7 @@
 #class-save-button{
     width: 2.5%;
 }
+.mat-form-field.class-edit-field{
```
This only needs to select the class, not `.mat-form-field`

floogulinc commented on src/app/class-component/class.component.html:
```diff
@@ -4,8 +4,8 @@
                 <mat-card-title flex id="decks-title" *ngIf="this.currentClass">
-                    <mat-form-field id="class-edit-field" *ngIf="editing">
-                        <input matInput *ngIf="this.canEdit" #input type="text" placeholder="Class Name" [(ngModel)]="this.currentClass.name">
+                    <mat-form-field class="class-edit-field" id="class-edit-field" *ngIf="editing">
```
We don't need both an ID and a class

floogulinc committed:
Refine and split renaming into its own component

Split the editable-text into its own component so it can be reused (eg in renaming decks (#20)). Also refines some styling to keep the spacing on the card overflow menu. Changed some CSS IDs to classes and cleaned up some other markup.

floogulinc commented:
We should merge this in first and use the new component in renaming decks (#20 and #89)

floogulinc committed:
Fix testing error with creation of EditableTextComponent

floogulinc reviewed (approved):
Looks good now!
